app/Relations/HasManyPhotosByTag.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace App\Relations;
+
+use App\Models\TagAlbum;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Collection;
+
+class HasManyPhotosByTag extends HasManyPhotos
+{
+	public function __construct(TagAlbum $owningAlbum)
+	{
+		parent::__construct($owningAlbum);
+	}
+
+	/**
+	 * Adds the constraints for single owning album to the base query.
+	 *
+	 * This method is called by the framework, if the photos of a
+	 * single tag albums are fetched.
+	 */
+	public function addConstraints(): void
+	{
+		if (static::$constraints) {
+			$this->addEagerConstraints([$this->owningAlbum]);
+		}
+	}
+
+	/**
+	 * Adds the constraints for a list of owning album to the base query.
+	 *
+	 * This method is called by the framework, if the related photos of a
+	 * list of owning albums are fetched.
+	 * The the unified result of the query is mapped to the specific albums
+	 * by {@link HasManyPhotosByTag::match()}.
+	 *
+	 * @param array $albums an array of {@link \App\Models\TagAlbum} whose photos are loaded
+	 */
+	public function addEagerConstraints(array $albums): void
+	{
+		if (count($albums) !== 1) {
+			throw new \InvalidArgumentException('eagerly fetching all photos of an album is only implemented for a single album at once');
+		}
+		/** @var TagAlbum $album */
+		$album = $albums[0];
+		$tags = explode(',', $album->show_tags);
```

Comment:
Maybe add a comment for future, if we want to add a DSL for a stronger tag parsing logic using conjunctive proposition, it should be here. :+1:

Reply:
What exactly do you want me to add as a comment? I am open to propositions.

Anyway, I actually hope that we either get rid of tag albums completely or if we cannot that we at least introduce a table `tags` and two proper m:n-relations between photos and tags and tags and tag albums.

At the moment matching photos to tag albums requires string operations on the PHP level and the SQL query for each tag album has an individual number of `WHERE`-clauses which is specific for the particular tag album. Hence, it is not possible to construct a single SQL query which fetches the photos for multiple tag albums.

This requires to fetch the photos for each tag album iteratively using individual SQL queries. It is impossible to eagerly load all photos for a set of tag albums at once. However, this would speed up the rendering of the root album if one has many tag albums.